### ISSUE TITLE
ConfirmDialogPreference feature

### ISSUE THREAD

**Issue by Senth** [michael-rapp/AndroidMaterialPreferences#14]
Hi Michael,

(Thanks for an awesome preference library! Found many along the way but this one was the best)

I have an feature request, my problem is that in some settings I have the ability to logout, or reset something. I.e. I only want a confirmation dialog preference. An example would be "Logout?" "Are you sure you want to logout?" With Logout / Cancel buttons.

I've extended your AbstractDialogPreference which makes this a lot easier, but it would be nice if more people can use it :) Of course you can change the code so that it matches your coding style :)

# Code
```java
import android.content.Context;
import android.support.annotation.NonNull;
import android.util.AttributeSet;

import de.mrapp.android.dialog.MaterialDialog;
import de.mrapp.android.preference.AbstractDialogPreference;

/**
 * A confirm dialog preference screen
 */
public class ConfirmDialogPreference extends AbstractDialogPreference {
private Listener mListener = null;

public ConfirmDialogPreference(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
	super(context, attrs, defStyleAttr, defStyleRes);
}

public ConfirmDialogPreference(Context context, AttributeSet attrs, int defStyleAttr) {
	super(context, attrs, defStyleAttr);
}

public ConfirmDialogPreference(Context context, AttributeSet attrs) {
	super(context, attrs);
}

public ConfirmDialogPreference(Context context) {
	super(context);
}

/**
 * Set listener for when the dialog is closed
 * @param listener listen to dialog events
 */
public void setListener(Listener listener) {
	mListener = listener;
}

@Override
protected boolean needInputMethod() {
	return false;
}

@Override
protected void onPrepareDialog(@NonNull MaterialDialog.Builder dialogBuilder) {
	// Does nothing
}

@Override
protected void onDialogClosed(boolean positiveResult) {
	if (mListener != null) {
		mListener.onDialogClosed(positiveResult);
	}
}

public interface Listener {
	/**
	 * Called when the dialog is closed
	 * @param positiveResult true if the user clicked the positive button
	 */
	void onDialogClosed(boolean positiveResult);
}
}
```

# Example
```java
ConfirmDialogPreference confirmDialogPreference = (ConfirmDialogPreference) findPreference(resources.getString(R.string.preference_logout_key));
confirmDialogPreference.setListener(new ConfirmDialogPreference.Listener() {
	@Override
	public void onDialogClosed(boolean positiveResult) {
		if (positiveResult) {
			user.logout();
		}
	}
});
```

**Comment by michael-rapp** [comment]
Hi,

thank you for your contribution. It really makes sense to include this into the library and your code is perfectly fine. However, I am thinking about adding the listener functionality to the already existing `AbstractDialogPreference` (maybe including even more listeners) so that other subclasses benefit from it as well. I will implement this soon.

**Comment by Senth** [comment]
:+1: Sounds like a good solution of the listener :)

**Comment by michael-rapp** [comment]
I have implemented the feature in version 2.6.0.

What I have done was to rename the class `AbstractDialogPreference` to `DialogPreference` and make it non-abstract. So this class can be used instead of creating a new one. Furthermore, I added various methods for attaching listeners to the preference's dialog (see release notes for full list). When using the  new `setOnClickListener(DialogInterface.OnClickListener)`-method, the attached listener is notified when one of the dialog's buttons is clicked. You can use it as follows:

```java
DialogPreference dialogPreference = (DialogPreference) findPreference(resources.getString(R.string.preference_logout_key));
dialogPreference.setOnClickListener(new DialogInterface.OnClickListener() {
    @Override
    public void onClick(DialogInterface dialog, int which) {
        if (which == DialogInterface.BUTTON_POSITIVE) {
            user.logout();
        }
    }
});
```
